### ISSUE TITLE
Parser does does not respect capabilities file when using `opa eval`

### ISSUE THREAD

**Issue by johanfylling** [open-policy-agent/opa#6379]
Given the policy:

```rego
package test

import future.keywords.if

p if {
   true
}
```

and a modified capabilities file, where we've removed the `if` keyword:

```json
{
  "builtins": [
    {
      "name": "eq",
      "decl": {
        "args": [
          {
            "type": "any"
          },
          {
            "type": "any"
          }
        ],
        "result": {
          "type": "boolean"
        },
        "type": "function"
      },
      "infix": "="
    }
  ],
  "future_keywords": [
    "contains",
    "every",
    "in"
  ],
  "features": [
    "rule_head_ref_string_prefixes"
  ]
}
```

When running `opa eval -d policy.rego --capabilities capabilities.json -fpretty 'data'`, we expect the error message:

```
1 error occurred: policy.rego:3: rego_parse_error: unexpected keyword, must be one of [contains every in]
	import future.keywords.if
```

but instead we get a successful evaluation: 

```json
{
  "test": {
    "p": true
  }
}
```

**Comment by johanfylling** [comment]
Is this a bug, or by design?

**Comment by johanfylling** [comment]
If this isn't a bug, and should be corrected, [this](https://github.com/open-policy-agent/opa/blob/eade10ae0e0d3436b146cc2c942387c26cbbfd5c/rego/rego.go#L1839C32-L1839C32) is where the capabilities are dropped.

**Comment by ashutosh-narkar** [comment]
> the reason for eval taking a --capabilities flag is to inform it about what built-ins are available, not to restrict its parsing.

If this was by design it should have been documented, no? Seems like a bug.

**Comment by johanfylling** [comment]
Yes, it's a peculiar limitation if by design. And I'd honestly be surprised if anyone expected `eval` to behave differently than `build` and `check`. Especially since it's undocumented.

**Comment by anderseknert** [comment]
I know I’m probably the only user of “opa parse”, but I guess it makes sense for that to have a capabilities flag added if that has an impact on parsing?

**Comment by johanfylling** [comment]
> it makes sense for that to have a capabilities flag added if that has an impact on parsing?

yes, I suppose so. All it'd do with the current set of capabilities we have is to stop it in its tracks and output an error. But it'd probably be a good thing to have it behave as the other commands.

**Comment by anderseknert** [comment]
Agreed, but fine to ignore until someone else asks about it.
